workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/SpatialOSBehaviourLibrary.cs

```diff
@@ -0,0 +1,187 @@
+            = "Reader or Writer found without a Component ID attribute, this is invalid!";
+
+        public SpatialOSBehaviourLibrary(ILogDispatcher logger)
+        {
+            this.logger = logger;
+        }
+
+        public void InjectAllReadersWriters(MonoBehaviour spatialOSBehaviour)
+        {
+            EnsureLoaded(spatialOSBehaviour.GetType());
+            if (invalidMonoBehaviourTypes.Contains(spatialOSBehaviour.GetType()))
+            {
+                return;
+            }
+
+            foreach (var readerWriterComponentId in componentReaderIdsForBehaviours[spatialOSBehaviour.GetType()])
+            {
+                Inject(spatialOSBehaviour, readerWriterComponentId);
+            }
+
+            foreach (var readerWriterComponentId in componentWriterIdsForBehaviours[spatialOSBehaviour.GetType()])
+            {
+                Inject(spatialOSBehaviour, readerWriterComponentId);
+            }
+        }
+
+        public void DeInjectAllReadersWriters(MonoBehaviour spatialOSBehaviour)
+        {
+            EnsureLoaded(spatialOSBehaviour.GetType());
+            if (invalidMonoBehaviourTypes.Contains(spatialOSBehaviour.GetType()))
+            {
+                return;
+            }
+
+            foreach (var readerWriterComponentId in componentReaderIdsForBehaviours[spatialOSBehaviour.GetType()])
+            {
+                DeInject(spatialOSBehaviour, readerWriterComponentId);
+            }
+
+            foreach (var readerWriterComponentId in componentWriterIdsForBehaviours[spatialOSBehaviour.GetType()])
+            {
+                DeInject(spatialOSBehaviour, readerWriterComponentId);
+            }
+        }
+
+        public List<uint> GetRequiredReaderComponentIds(Type behaviourType)
+        {
+            EnsureLoaded(behaviourType);
+            return componentReaderIdsForBehaviours[behaviourType];
+        }
+
+        public List<uint> GetRequiredWriterComponentIds(Type behaviourType)
+        {
+            EnsureLoaded(behaviourType);
+            return componentWriterIdsForBehaviours[behaviourType];
+        }
+
+        private void Inject(MonoBehaviour spatialOSBehaviour, uint componentId)
+        {
+            var readerWriter = ReaderWriterFactory.CreateReaderWriter(componentId);
+            var memberAdapter = adapterCache[spatialOSBehaviour.GetType()][componentId];
+            memberAdapter.SetValue(spatialOSBehaviour, readerWriter);
+        }
+
+        private void DeInject(MonoBehaviour spatialOSBehaviour, uint componentId)
+        {
+            var memberAdapter = adapterCache[spatialOSBehaviour.GetType()][componentId];
+            memberAdapter.SetValue(spatialOSBehaviour, null);
+        }
+
+        private void EnsureLoaded(Type behaviourType)
+        {
+            if (adapterCache.ContainsKey(behaviourType))
+            {
+                return;
+            }
+
+            var adapters = GetMembersWithMatchingAttributes(behaviourType);
```

Comment:
These are no longer adapters, so the variable name of `adapters` is not as suitable

Reply:
See other `adapter`~ variable names in the PR